test/ui-testing/tags-crud.js

```diff
@@ -0,0 +1,131 @@
+/* global describe, it, before, after, Nightmare */
+const AgreementCRUD = require('./agreement-crud');
+
+module.exports.test = (uiTestCtx) => {
+  const number = Math.round(Math.random() * 100000);
+  const testTag = `test${Math.floor(Math.random() * 10000)}`;
+  const agreement = {
+    name: `Linked License Agreement #${number}`,
+  };
+
+  describe('Login > Enable Tags > Find user > Create Tags > Logout\n', function test() {
+    const { config, helpers } = uiTestCtx;
+    const nightmare = new Nightmare(config.nightmare);
+    nightmare.options.width = 1000; // added this temporarily as MultiSelect doesnt work well with narrow screen sizes
+
+    this.timeout(Number(config.test_timeout));
+
+    describe('login > create license > edit terms > create agreement > link license > check term > logout', () => {
+      before((done) => {
+        helpers.login(nightmare, config, done);
+      });
+
+      after((done) => {
+        helpers.logout(nightmare, config, done);
+      });
+
+      it('should Enable tags in settings', done => {
+        nightmare
+          .wait(config.select.settings)
+          .click(config.select.settings)
+          .wait('#app-list-item-clickable-settings')
+          .wait('a[href="/settings/tags"]')
+          .click('a[href="/settings/tags"]')
+          .wait('a[href="/settings/tags/general"]')
+          .click('a[href="/settings/tags/general"]')
+          .wait('#tags_enabled')
+          .wait(222)
+          .evaluate(() => {
+            const element = document.querySelector('[value="true"]');
+            element.click();
+          })
+          .then(() => {
+            nightmare
+              .wait(222)
+              .wait('#tags_enabled')
+              .click('#tags_enabled')
+              .wait('#clickable-save-config')
+              .click('#clickable-save-config')
+              .then(done)
+              .catch(done);
+          });
+      });
+
+      it('should open Agreements app', done => {
+        helpers.clickApp(nightmare, done, 'agreements');
+      });
+
+      it('should confirm correct URL', done => {
+        nightmare
+          .evaluate(() => document.location.pathname)
+          .then(pathName => {
+            if (!pathName.includes('/erm/agreements')) throw Error('URL is incorrect');
+            done();
+          })
+          .catch(done);
+      });
+
+      it('should create agreement', done => {
+        AgreementCRUD.createAgreement(nightmare, done, agreement);
+      });
+
+      it(`should search for and find agreement: ${agreement.name}`, done => {
```

Comment:
I've got no issues with you being extra cautious in tests and having specific cases for things that aren't integral to the _real_ thing we're testing (tags in this case), but I tend not to because:
* I'm lazy
* It's already tested in another place, eg `agreement-crud`
* It slows down running the tests

Up to you, just letting you know for future use.